libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -0,0 +1,443 @@
+
+    def teardown(self):
+        self.log('Teardown', show=True)
+        # Kill VM        
+        self._virsh_call(['destroy', self.vm_name])
+        # Remove VM
+        self._virsh_call(['undefine', self.vm_name, '--remove-all-storage'], check=True)
+        
+        self.vm_semaphore.release()
+
+        with self.setup_lock:
+            # If this instance is the last one using temp dir -> remove it
+            self.instances[self.git_anaconda_branch].remove(self)
+            if len(self.instances[self.git_anaconda_branch]) == 0:
+                self.temp_dirs[self.git_anaconda_branch].cleanup()
+                
+                del self.instances[self.git_anaconda_branch]
+                # Remove container image, if this was the last anacoda-webui workflow
+                if self.instances == dict():
+                    self.container.remove_image()
+
+    def terminate(self):
+        # Kill VM
+        self._virsh_call(['destroy', self.vm_name])
+
+    def log(self, message, name='workflow', show=False):
+        if name == 'workflow' and show:
+            self.last_log = message
+        super().log(message, name)
+
+    def displayStatus(self):
+        return self.last_log
+
+    def _start_vm(self):
+        """Prepare virt-install command, start VM and wait for IP address
+
+        :return: IP address
+        :rtype: str
+        """
+        self.log('Starting VM', show=True)
+
+        # Get compose location
+        os_url = self.installation_source.repos[self.installation_source.base_repo][self.architecture]
+        if self.installation_source.kernel_path and self.installation_source.initrd_path:
+            location = f'{os_url},kernel={self.installation_source.kernel_path},initrd={self.installation_source.initrd_path}'
+        else:
+            location = os_url
+
+        # Construct kernel cmdline for --extra-args
+        kernel_cmdline = f'inst.sshd inst.webui inst.webui.remote inst.stage2={self.os_url}'
+        if self.additional_repos:
+            for repo in self.additional_repos:
+                kernel_cmdline += f' inst.addrepo={self.installation_source.repo[repo][self.architecture]}'
+
+        if self.kernel_cmdline:
+            kernel_cmdline += f' {self.kernel_cmdline}'
+
+        # Assemble virt-install command
+        cmd = ['virt-install', '--connect', self.hypervisor_host, '--noautoconsole',
+            '-n', self.vm_name, '--os-variant', 'rhel-unknown', '--location', location,
+            '--memory', '4096', '--vcpus', '2', '--disk', 'size=10',
+            '--extra-args', kernel_cmdline]
+
+        LOGGER.info('Running: ' + repr(cmd))
+        process = subprocess.run(cmd ,check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        self.log(process.stdout.decode())
+
+        # Wait for machine to start and get IP
+        self.log('Waiting for IP', show=True)
+        startup_timeout = 120 + time.time()
+        while True:
```

Comment:
Please move this block along with the return statement to a separate method/function.